### PR TITLE
refactor: Simplify code

### PR DESCRIPTION
**Pull request by aragon999**
### 1. Why is this change necessary?
These are some changes which accumulated over time, when I was going through the code.

### 2. What does this change do, exactly?
It contains mostly cleanups and simplifications, which only should improve the developer experience and make use of new PHP features.

Not sure if such a PR is appreciated or not, or if it should be split into more smaller PRs.
Also I am not sure if a changelog is required for these changes.

### 3. Describe each step to reproduce the issue or behaviour.
Look at the code.

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Administration/Controller/UserConfigController.php [hunk 1]
```diff
@@ -73,7 +81,7 @@ private function getOwnUserConfig(Context $context, array $keys): array
             $criteria->addFilter(new EqualsAnyFilter('key', $keys));
         }
 
-        return $this->userConfigRepository->search($criteria, $context)->getElements();
+        return $this->userConfigRepository->search($criteria, $context)->getEntities()->getElements();
```
Comment (mitelg) [inline src/Administration/Controller/UserConfigController.php]: instead of returning the array, I would prefer returning the collection

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php [hunk 2]
```diff
@@ -21,26 +22,36 @@ class ProductFeatureBuilder
...
+     * @param EntityRepository<CustomFieldCollection> $customFieldRepository
      */
     public function __construct(
         private readonly EntityRepository $customFieldRepository,
         private readonly LanguageLocaleCodeProvider $languageLocaleProvider
     ) {
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
Comment (mitelg) [inline src/Core/Content/Product/Cart/ProductFeatureBuilder.php]:
```suggestion
     * @param iterable<LineItem> $lineItems
```

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php [hunk 3]
```diff
@@ -21,26 +22,36 @@ class ProductFeatureBuilder
...
+     */
     public function prepare(iterable $lineItems, CartDataCollection $data, SalesChannelContext $context): void
     {
         $this->loadCustomFields($lineItems, $data, $context);
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
Comment (mitelg) [inline src/Core/Content/Product/Cart/ProductFeatureBuilder.php]:
```suggestion
     * @param iterable<LineItem> $lineItems
```

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php [hunk 4]
```diff
@@ -94,23 +101,25 @@ private function buildFeatures(CartDataCollection $data, LineItem $lineItem, Sal
         return array_filter($features);
     }
 
+    /**
+     * @param LineItem[] $lineItems
```
Comment (mitelg) [inline src/Core/Content/Product/Cart/ProductFeatureBuilder.php]:
```suggestion
     * @param iterable<LineItem> $lineItems
```

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php [hunk 5]
```diff
@@ -162,6 +170,9 @@ private function isRequiredCustomField(string $name, SalesChannelProductEntity $
         return false;
     }
 
+    /**
+     * @return array{label: string, value: array<mixed>, type: string}
```
Comment (mitelg) [inline src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: is value really an array :thinking: 

```suggestion
     * @return array{label: string, value: mixed, type: string}
```
I guess? :thinking:

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php [hunk 6]
```diff
@@ -240,9 +256,10 @@ private function getCustomField(string $name, CartDataCollection $data, SalesCha
         }
 
         $customField = $data->get($fieldKey);
-        $label = $this->getCustomFieldLabel($customField);
+        \assert($customField instanceof CustomFieldEntity);
```
Comment (mitelg) [inline src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: maybe do a real if condition instead :thinking: or can we really be sure, this is always true?

File: src/Core/Content/Test/ProductExport/Service/ProductExporterTest.php [hunk 7]
```diff
@@ -58,36 +56,12 @@ public function testExport(): void
...
-
-        static::assertIsString($fileContent);
-        $csvRows = explode(\PHP_EOL, $fileContent);
 
         static::assertTrue($this->fileSystem->directoryExists($this->getContainer()->getParameter('product_export.directory')));
         static::assertTrue($this->fileSystem->fileExists($filePath));
-        static::assertCount(4, $csvRows);
-    }
-
-    public function testExportPagination(): void
```
Comment (mitelg) [inline src/Core/Content/Test/ProductExport/Service/ProductExporterTest.php:71]: why did you remove this test?

**Review by mitelg — CHANGES_REQUESTED** [review]
really nice! thank you very much for that :+1: I like such work :grin:

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php (hunk 6 above)
Comment (aragon999) [reply src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: Mh, so the data key is `custom-field-{$customFieldName}` and it has been checked that is exists. So it actually should always be a `CustomFieldEntity` otherwise something is broken.

File: src/Core/Content/Test/ProductExport/Service/ProductExporterTest.php (hunk 7 above)
Comment (aragon999) [reply src/Core/Content/Test/ProductExport/Service/ProductExporterTest.php:71]: As it is actually the same as the `testExport()` when the "useles" assert is removed :-)

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php (hunk 5 above)
Comment (aragon999) [reply src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: Yup, missed that :see_no_evil:

**Comment by aragon999** [comment]
@mitelg Thank you :-) Yes from time to time we need to cleanup :-D 

I adjusted the obvious changes, and added comments to the other ones. Additionally I also had some changes for the `ShippingMethodValidator` and the `CleanPersonalDataCommand`. I added them in a new commit (but I can also fixup them, if desired).

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php (hunk 6 above)
Comment (mitelg) [reply src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: if condition that then throws an exception maybe?

**Comment by mitelg** [comment]
now some tests are failing, because of the changed exception messages :wink:

File: src/Core/Content/Product/Cart/ProductFeatureBuilder.php (hunk 6 above)
Comment (aragon999) [reply src/Core/Content/Product/Cart/ProductFeatureBuilder.php]: Mh, okay :-D I changed the behavior to throw an exception.

**Comment by aragon999** [comment]
Let's wait for the pipeline, but I hope now the tests are fixed (there was actually a bug in my changes :-) )

**Review by mitelg — APPROVED** [review]

**Review by cyl3x — APPROVED** [review]

**Comment by cyl3x** [comment]
@aragon999 Thanks for your contribution! 🎉
